### PR TITLE
Rename markdown filenames

### LINKED ISSUE #799: Rename improper markdown filenames
In 'How to Contribute', we guide contributors to set markdown filename with this rule:
> no capitalization and no space, and .md at the end

As a result of guidance and compliance of contributors, most of markdown filenames are being appropriate.

But there exist some exceptions, so I suggest to rename those.

I am suggesting this because our focus should be on awareness and getting people to use and **reference the terms** (borrowing @CathPag's words).

[Example]
- `TLS(Transport Layer Security).md`
  - image
    (which seems not great, IMO)
  - Suggestion: `transport-layer-security.md`

---

[Suggestions]

| As-is  | To-be  | Current URL  |
|---|---|---|
| `TLS(Transport Layer Security).md`  | `transport_layer_security.md`  | https://glossary.cncf.io/tlstransport-layer-security/  |
| `container-image.md`  | `container_image.md`  | https://glossary.cncf.io/container-image/ <br> For now this term is not displayed, since `status` is not `Completed`  |
| `mTLS (Mutual Transport Layer Security).md`  | `mutual_transport_layer_security.md`  | https://glossary.cncf.io/mtls-mutual-transport-layer-security/  |
| `versioncontrol.md`  | `version_control.md`  | https://glossary.cncf.io/versioncontrol/  |
|   |   |   |
|   |   |   |
|   |   |   |
|   |   |   |
|   |   |   |
|   |   |   |
|   |   |   |
|   |   |   |
|   |   |   |
|   |   |   |

---

[Caution]

When updating filenames, one should also update the links toward those terms,
and the changes should be propagated to each L10n directories/branches too.

### PR DESCRIPTION
**Pull request by jihoon-seo**
This PR
- closes #799.
- renames markdown filenames, replacing all the underscores (`_`) with hyphen (`-`).
  - "no capitalization, no spaces, no parentheses, and use hyphens to separate words"

---

[Commits]
- Insert empty lines before and after the header lines (for all languages)
- Add semantic line breaks (for English files only)
- Rename markdown filenames (for all languages)
- Update relevant links (for all languages)
- ...

### CODE REVIEW

**Review by iamNoah1 — CHANGES_REQUESTED** [review]
Commits seem not be signed correctly

**Comment by raelga** [comment]
ack!

**Comment by jihoon-seo** [comment]
@iamNoah1 As per your comment, I added `Signed-off-by:` string to all 4 commits' messages. 😊

**Review by raelga — APPROVED** [review]

**Review by iamNoah1 — APPROVED** [review]

File: content/hi/devops.md [hunk 1]
```diff
@@ -6,12 +6,15 @@ exclude_search: true
...
 
 ## समस्या
-परंपरागत रूप से, [टाइटली-कपल्ड](/tightly_coupled_architectures/) [मोनोलिथिक ऐप्स](/monolith_apps/) का प्रयोग वाले जटिल संगठनों में, काम आम तौर पर कई समूहों के बीच खंडित होता था। इसमें एप्लीकेशन कई टीमों के बीच जाता था और काफी लम्बा समय लगता था। हर बार जब कोई घटक या अपडेट तैयार होता था, तो उसे अगली टीम के लिए एक कतार में रखा जाता था। क्योंकि प्रत्येक व्यक्ति केवल परियोजना के एक छोटे से हिस्से पर काम करता था, इस दृष्टिकोण के कारण स्वामित्व की कमी होती थी। उनका लक्ष्य काम को अगले समूह तक पहुंचाना था, न कि ग्राहक को सही कार्यक्षमता प्रदान करना - प्राथमिकताओं का एक स्पष्ट गलत संरेखण।
+
+परंपरागत रूप से, [टाइटली-कपल्ड](/tightly-coupled-architectures/) [मोनोलिथिक ऐप्स](/monolith-apps/) का प्रयोग वाले जटिल संगठनों में, काम आम तौर पर कई समूहों के बीच खंडित होता था। इसमें एप्लीकेशन कई टीमों के बीच जाता था और काफी लम्बा समय लगता था। हर बार जब कोई घटक या अपडेट तैयार होता था, तो उसे अगली टीम के लिए एक कतार में रखा जाता था। क्योंकि प्रत्येक व्यक्ति केवल परियोजना के एक छोटे से हिस्से पर काम करता था, इस दृष्टिकोण के कारण स्वामित्व की कमी होती थी। उनका लक्ष्य काम को अगले समूह तक पहुंचाना था, न कि ग्राहक को सही कार्यक्षमता प्रदान करना - प्राथमिकताओं का एक स्पष्ट गलत संरेखण।
```
Comment (seokho-son) [inline content/hi/devops.md:14]:
```suggestion
परंपरागत रूप से, [टाइटली-कपल्ड](/tightly-coupled-architectures/) [मोनोलिथिक ऐप्स](/monolithic-apps/) का प्रयोग वाले जटिल संगठनों में, काम आम तौर पर कई समूहों के बीच खंडित होता था। इसमें एप्लीकेशन कई टीमों के बीच जाता था और काफी लम्बा समय लगता था। हर बार जब कोई घटक या अपडेट तैयार होता था, तो उसे अगली टीम के लिए एक कतार में रखा जाता था। क्योंकि प्रत्येक व्यक्ति केवल परियोजना के एक छोटे से हिस्से पर काम करता था, इस दृष्टिकोण के कारण स्वामित्व की कमी होती थी। उनका लक्ष्य काम को अगले समूह तक पहुंचाना था, न कि ग्राहक को सही कार्यक्षमता प्रदान करना - प्राथमिकताओं का एक स्पष्ट गलत संरेखण।
```

File: content/zh-cn/scalability.md [hunk 2]
```diff
@@ -6,4 +6,4 @@ category: 属性
 
 可扩展性指的是一个系统能有多大的发展。这就是增加做任何系统应该做的事情的能力。 例如，[Kubernetes](/zh-cn/kubernetes/) [集群](/zh/cluster/) 通过增加或减少 [容器化](/zh-cn/containerization/) 应用程序的数量来进行扩展，但这种可扩展性取决于几个因素。 它有多少[节点](/nodes/)，每个节点可以处理多少个[容器](/zh-cn/container/)，控制平面可以支持多少条记录和操作？
```
Comment (seokho-son) [inline content/zh-cn/scalability.md:7]:
```suggestion
可扩展性指的是一个系统能有多大的发展。这就是增加做任何系统应该做的事情的能力。 例如，[Kubernetes](/zh-cn/kubernetes/) [集群](/zh-cn/cluster/) 通过增加或减少 [容器化](/zh-cn/containerization/) 应用程序的数量来进行扩展，但这种可扩展性取决于几个因素。 它有多少[节点](/nodes/)，每个节点可以处理多少个[容器](/zh-cn/container/)，控制平面可以支持多少条记录和操作？
```

**Review by seokho-son — COMMENTED** [review]
Hi @jihoon-seo 

I found two broken links.

**Comment by jihoon-seo** [comment]
@seokho-son Fixed! Thank you for your review. 😊

**Comment by seokho-son** [comment]
Hi, l10n team reviewers !

I hope to expedite this PR. (including lots of changes)

So, I hope to coordinate reviewers for localized contents. (Please review or assign another reviewer from your team)
 - Check (missing, problematic, inaccurate, ..) changed files for your language. 
   - (https://github.com/cncf/glossary/pull/922/files content/ko/* for instance)
 - Check preview for your language. 
   - (https://deploy-preview-922--cncfglossary.netlify.app/ko for Korean for instance)


--- 

- content/bn/* : @mitul3737 
- content/hi/* : @anubha-v-ardhan
- contnet/zh-cn/* : @Jacob953
- contnet/ko/* : @yunkon-kim 
- content/it/* : @sistella 
- content/de/* : seems reviewed already :)
- content/es/* : seems reviewed already :)

---

- content/en/* : @seokho-son @iamNoah1

**Review by jihoon-seo — COMMENTED** [review]
I will remove the hardcoded domain name (`glossary.cncf.io`)
in `content/ko/virtual-machine.md` too.

```diff
-[가상화(virtualization)](https://glossary.cncf.io/virtualization/)
+[가상화(virtualization)](/virtualization/)

-[베어 메탈(bare metal)](https://glossary.cncf.io/bare_metal_machine/)
+[베어 메탈(bare metal)](/bare-metal-machine/)
```

**Review by mitul3737 — APPROVED** [review]
LGTM FOR Bengali Contents

**Review by iamNoah1 — APPROVED** [review]
Big round of applause @jihoon-seo for this huge edit :). Thank you. English content looks good :) 

lgtm

**Comment by CathPag** [comment]
Thank you, @jihoon-seo!!!

**Comment by seokho-son** [comment]
LGTM for Korean contents

**Review by Jacob953 — APPROVED** [review]
LGTM for Chinese content.

**Comment by seokho-son** [comment]
Hello, reviewers :) 
Most of l10n teams confirmed the changes of this PR. Since this PR touches many files, I hope to avoid conflicts with other coming PRs.

I'v checked (reviewed) files and previews for the l10n files which are not reviewed by l10n team yet. LGTM.

Considering a time limit, let me approve this PR .

Hi @cjyabraham . This PR will make conflicts to your PR #930 . (Both this PR and PR #930 changes overall files)
I'd like to avoid conflicts but there is no way this time. Thank you for the website enhancement !

**Comment by jihoon-seo** [comment]
@seokho-son Please give me some time to tidy commits and apply [this](https://github.com/cncf/glossary/pull/922#pullrequestreview-994866484),
I will handle these tomorrow soon. :)

**Comment by iamNoah1** [comment]
Go for it @seokho-son. Conflicts can not be avoided sometimes, especially with so many changes. The sooner we get this merged, the better. We can fix broken things afterwards :)

**Comment by jihoon-seo** [comment]
@seokho-son Done! Please take a look. Thank you. 😊

**Review by seokho-son — APPROVED** [review]
Final LGTM !

**Comment by iamNoah1** [comment]
Awesome :)

**Pull request merged**